src/components/forms/Checkbox.jsx

```diff
@@ -11,62 +11,89 @@ import CheckIndeterminateIcon from 'lib/icons/react/CheckIndeterminateIcon';
+    const checked = rest.checked != null ? rest.checked : checkedState;
 
-  const IconUnchecked = CheckEmptyIcon;
-  const IconChecked = indeterminate ? CheckIndeterminateIcon : CheckFilledIcon;
+    // We should maybe fire onChange in "addition" to setChecked, based on whether we
+    // interpret checkbox as being "controlled" or not
+    const onChange = rest.onChange ? rest.onChange : (e) => setChecked(e.target.checked);
 
-  return (
-    <div
-      className={cx('fdsCheckable fdsCheckbox', { 'fdsCheckable--disabled': disabled })}
-    >
-      <input
-        type="checkbox"
-        id={id}
-        className="media--xs"
-        disabled={disabled}
-        ref={inputRef}
-        {...rest}
-      />
-      <label className="flush--bottom" htmlFor={id}>
-        <span className="fdsCheckable-icon--checked">
-          <IconChecked size="xs" />
-        </span>
-        <span className="fdsCheckable-icon--unchecked">
-          <IconUnchecked size="xs" />
-        </span>
-        {showLabel && <span className="padding--left--half">{label}</span>}
-      </label>
-    </div>
-  );
-};
+    const CheckboxIcon = (() => {
+      if (checked) {
+        if (checked === 'mixed') {
+          return CheckIndeterminateIcon;
+        }
+        return CheckFilledIcon;
+      }
+      return CheckEmptyIcon;
+    })();
 
-Checkbox.propTypes = {
-  /** Label used for a11y attributes _and_ the rendered `label` element */
-  label: PropTypes.string.isRequired,
+    const Element = showLabel ? 'label' : 'div';
 
-  /** If the supplied `label` prop should be rendered to the screen. */
-  showLabel: PropTypes.bool,
+    const checkboxColor = () => {
+      if (disabled) return FDS.COLOR_GRAY;
+      if (checked) return FDS.COLOR_BLUE;
+      return FDS.FONT_COLOR_SECONDARY;
+    };
+
+    return (
+      <Element
+        className={cx('fdsCheckable fdsCheckbox', 'flush--bottom', {
+          'fdsCheckable--disabled': disabled,
+        })}
+      >
+        <input
+          ref={ref}
+          type="checkbox"
+          className="media--xs"
+          disabled={disabled}
+          onChange={onChange}
+          {...rest}
+        />
+        <span
+          aria-hidden
+          className="fdsCheckable--icon"
+          style={{ pointerEvents: 'none' }}
+        >
+          <CheckboxIcon size="xs" color={checkboxColor()} />
+        </span>
+        {showLabel && (
+          <span
+            className={cx('padding--left--half', {
+              'color--secondary': disabled,
+            })}
+          >
+            {label}
+          </span>
+        )}
+      </Element>
+    );
+  }
+);
+
+Checkbox.displayName = 'Checkbox';
 
-  /** Ref for input element */
-  inputRef: PropTypes.func,
+Checkbox.propTypes = {
+  /**
+   * Setting any one of the 3 values here will turn component into controlled component.
+   * */
+  checked: PropTypes.oneOf([true, false, 'mixed']),
```

Comment:
Designers and the html spec calls this state `indeterminate`. What do you think about using that instead of `mixed`?